inc/class-gutenberg-ramp-criteria.php

```diff
@@ -74,6 +80,55 @@ public function set( $criteria ) {
 		return false;
 	}
 
+	/**
+	 * Merge all criteria from all calls to ramp_for_gutenberg_load_gutenberg()
+	 *
+	 * @param array $criteria The new criteria to be merged with existing criteria.
+	 * @return array The merged criteria.
+	 */
+	public function merge( $criteria = [] ) {
+
+		// Get our existing criteria
+		$existing_criteria = self::$criteria;
+
+		// If this hasn't been set to active yet, there's nothing to merge - theoretically this should never be triggered.
+		$ramp = Gutenberg_Ramp::get_instance();
+		if ( ! $ramp->active ) {
+			return $criteria;
+		}
+
+		$merged_criteria = [];
+
+		/**
+		 * For the current setup of criteria, we can only merge 'post_ids' and 'post_types'.
+		 * The 'load' parameter only accepts a single value, so nothing to merge and the _last_ value it's passed will be the one used.
+		 */ 
+		if ( is_array( $existing_criteria ) ) {
```

Comment:
I suppose I was just being safe here in checking to make sure `$existing_criteria` was an array. There's no _normal_ scenario where it wouldn't be an array, but I just never trust users enough to not check the type before I run a `foreach`. Happy to remove the `is_array()` check, though.

Reply:
Never trusting the input is good, so I don't think is_array should be removed. But I think that we can work around that. So if there are no existing criteria, maybe we can just return the new criteria instantly, instead of trying to merge an empty array. Maybe the same goes for the other way around - when there are no new criteria to be merged? That's just an idea -  probably needs more thought in-code.

Reply:
Gotcha. On both `$existing_criteria` and `$criteria`, if they're not an array or they're empty and return the other. I'll have a PR in a few minutes for everything. 👍